test/e2e/e2e_test.go

```diff
@@ -121,57 +112,55 @@ func TestE2E(t *testing.T) {
 		zone := testContext.CloudConfig.Zone
 		region, err := gcecloud.GetGCERegion(zone)
 		if err != nil {
-			glog.Fatalf("error parsing GCE region from zone %q: %v", zone, err)
+			return fmt.Errorf("error parsing GCE/GKE region from zone %q: %v", zone, err)
 		}
 		managedZones := []string{zone} // Only single-zone for now
 		cloudConfig.Provider, err = gcecloud.CreateGCECloud(testContext.CloudConfig.ProjectID, region, zone, managedZones, "" /* networkUrl */, tokenSource, false /* useMetadataServer */)
 		if err != nil {
-			glog.Fatal("Error building GCE provider: ", err)
+			return fmt.Errorf("Error building GCE/GKE provider: ", err)
 		}
 
-	}
-
-	if testContext.Provider == "aws" {
+	case "aws":
 		awsConfig := "[Global]\n"
 		if cloudConfig.Zone == "" {
-			glog.Fatal("gce-zone must be specified for AWS")
+			return fmt.Errorf("gce-zone must be specified for AWS")
 		}
 		awsConfig += fmt.Sprintf("Zone=%s\n", cloudConfig.Zone)
 
 		if cloudConfig.ClusterTag == "" {
-			glog.Fatal("--cluster-tag must be specified for AWS")
+			return fmt.Errorf("--cluster-tag must be specified for AWS")
 		}
 		awsConfig += fmt.Sprintf("KubernetesClusterTag=%s\n", cloudConfig.ClusterTag)
 
 		var err error
 		cloudConfig.Provider, err = cloudprovider.GetCloudProvider(testContext.Provider, strings.NewReader(awsConfig))
 		if err != nil {
-			glog.Fatal("Error building AWS provider: ", err)
+			return fmt.Errorf("Error building AWS provider: ", err)
 		}
-	}
 
-	// Disable skipped tests unless they are explicitly requested.
-	if config.GinkgoConfig.FocusString == "" && config.GinkgoConfig.SkipString == "" {
-		// TODO(ihmccreery) Remove [Skipped] once all [Skipped] labels have been reclassified.
-		config.GinkgoConfig.SkipString = `\[Flaky\]|\[Skipped\]|\[Feature:.+\]`
 	}
-	gomega.RegisterFailHandler(ginkgo.Fail)
 
-	c, err := loadClient()
-	if err != nil {
-		glog.Fatal("Error loading client: ", err)
-	}
+	return nil
+}
+
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
```

Comment:
This is a wonky enough syntax and procedure that I think it deserves a comment.  What are you doing here & why?